apps/backend/src/modules/users/user-details.repository.ts

```diff
@@ -0,0 +1,47 @@
+import { type Repository } from "~/libs/types/types.js";
+import { UserDetailsEntity } from "~/modules/users/user-details.entity.js";
+import { type UserDetailsModel } from "~/modules/users/user-details.model.js";
+
+class UserDetailsRepository implements Repository {
+	private userDetailsModel: typeof UserDetailsModel;
+
+	public constructor(userDetailsModel: typeof UserDetailsModel) {
+		this.userDetailsModel = userDetailsModel;
+	}
+
+	public async create(entity: UserDetailsEntity): Promise<UserDetailsEntity> {
+		const { firstName, lastName, userId } = entity.toNewObject();
+
+		const userDetails = await this.userDetailsModel
+			.query()
+			.insert({
+				firstName,
+				lastName,
+				userId,
+			})
+			.returning("*")
+			.execute();
+
+		return UserDetailsEntity.initialize(userDetails);
+	}
+
+	public delete(): ReturnType<Repository["delete"]> {
+		return Promise.resolve(true);
+	}
+
+	public find(): ReturnType<Repository["find"]> {
+		return Promise.resolve(null);
+	}
```

Comment:
Do not add placeholders, add real logic. Delete and find by id logic should be implemented